src/hotspot/share/memory/metadataFactory.hpp

```diff
@@ -65,10 +67,17 @@ class MetadataFactory : AllStatic {
     if (md != NULL) {
       assert(loader_data != NULL, "shouldn't pass null");
       int size = md->size();
-      // Call metadata's deallocate function which will call deallocate fields
+      // Call metadata's deallocate function which will deallocate fields and release_C_heap_structures
       assert(!md->on_stack(), "can't deallocate things on stack");
       assert(!md->is_shared(), "cannot deallocate if in shared spaces");
       md->deallocate_contents(loader_data);
+      // If requested, call the destructor. This is currently used for MethodData which has a member
+      // that needs to be destructed to release resources. Most Metadata derived classes have noop
+      // destructors and/or cleanup using deallocate_contents.
+      // T is a potentially const or volatile qualified pointer. Remove the pointer and any const
+      // or volatile so we can call the destructor of the type T points to.
+      using U = typename RemoveCV<typename RemovePointer<T>::type>::type;
```

Comment:
Use `std::remove_pointer<T>` and `std::remove_cv<T>` from `<type_traits>` in
preference to our pre-C++11/14 workarounds. You can also avoid `typename` and
`::type` by using the "_t"-suffixed versions. So
```
using U = std::remove_cv_t<std::remove_pointer_t<T>>;
```
And "If requested" in the comment? That suggests some conditionalization, of
which there is none. Maybe something like "Most of the cleanup is done via
deallocate_contents, but some types require some additional cleanup via the
destructor."

Reply:
I fixed this as suggested.  Thanks.